File: Low Level Design (LLD)/Problems/05. FoodDeliverySystem/Java/service/PaymentService.java
package service;

import payment.Payment;
import utils.PaymentStatus;

import java.util.HashMap;
import java.util.Map;

public class PaymentService {
    private Map<String, Payment> payments = new HashMap<>();

    public void processPayment(Payment payment) {
        payments.put(payment.getId(), payment);
        System.out.println("Payment processing...\nDone!");
        payment.setStatus(PaymentStatus.COMPLETED);
    }

    public Payment getPaymentById(String id) {
        return payments.get(id);
    }
}

File: Low Level Design (LLD)/Problems/05. FoodDeliverySystem/Cpp/service/src/PaymentService.cpp
#include "../include/PaymentService.h"
#include <iostream>

void PaymentService::processPayment(Payment payment) {
    payments[payment.getId()] = payment;
    std::cout << "Payment processing...\nDone!" << std::endl;
    payments[payment.getId()].setStatus(PaymentStatus::COMPLETED);
}

const Payment* PaymentService::getPaymentById(const std::string& id) const {
    auto it = payments.find(id);
    if (it != payments.end()) {
        return &it->second;
    }
    return nullptr;
}